google/cloud/spanner/client_test.cc

```diff
@@ -386,10 +387,10 @@ TEST(ClientTest, MakeConnectionOptionalArguments) {
   auto conn = MakeConnection(db);
   EXPECT_NE(conn, nullptr);
 
-  conn = MakeConnection(db, ConnectionOptions());
+  conn = MakeConnection(db, internal::Options{});
   EXPECT_NE(conn, nullptr);
 
-  conn = MakeConnection(db, ConnectionOptions(), SessionPoolOptions());
+  conn = MakeConnection(db, internal::Options{}, SessionPoolOptions{});
```

Comment:
I wonder if `SessionPoolOptions` can be subsumed into `Options{}`, we added it because it was not possible to easily add stuff to `ConnectionOptions`.

Reply:
Yes, I think it can. Good idea. Likely also the retry and backoff policies. I think we'll be able to get the `MakeConnection` factories to look like `MakeConnection(Database, Options opts = {})` with no other overloads.